### DISCUSSION TITLE
For Control nodes, we should add Custom Max Size

### DISCUSSION THREAD

**Discussion by philippedcote** [godotengine/godot-proposals#7597] [GUI]
While working in the Editor, Control nodes have a tendency to grow to random proportions for unknown reasons.

We think adding a Custom Max Size next to Custom Min Size would greatly help manage those problems.

Thanks.

**Comment by Calinou** [comment]
This was previously proposed in https://github.com/godotengine/godot-proposals/issues/1802 and was rejected.

**Reply by philippedcote** [reply]
Thanks for pointing that out.

I just went to read about the arguments against, but the problem persists: Control nodes have a tendency to resize themselves randomly for no apparent reasons. We don't have a clue of what is causing these problems to happen. Until the core problem is fixed, we need a band-aid solution. Any clues?

By the way, Unity Editor allow the user to set up a "Preferred Width", which kinda works like a Max Width. Maybe this could be a solution to explore.

**Comment by djrain** [comment]
There is a bug involving AnimationPlayer that can cause Control nodes to double in size, not sure if that might be your issue.

**Reply by philippedcote** [reply]
Yes, it could be related to AnimationPlayer, indeed.
